components/kyma-environment-broker/internal/process/deprovisioning/check_kyma_resource_deleted_step.go

```diff
@@ -0,0 +1,57 @@
+package deprovisioning
+
+import (
+	"context"
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process/steps"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+)
+
+type CheckKymaResourceDeletedStep struct {
+	operationManager *process.OperationManager
+	kcpClient        client.Client
+}
+
+func NewCheckKymaResourceDeletedStep(operations storage.Operations, kcpClient client.Client) *CheckKymaResourceDeletedStep {
+	return &CheckKymaResourceDeletedStep{
+		operationManager: process.NewOperationManager(operations),
+		kcpClient:        kcpClient,
+	}
+}
+
+func (step *CheckKymaResourceDeletedStep) Name() string {
+	return "Check_Kyma_Resource_Deleted"
+}
+
+func (step *CheckKymaResourceDeletedStep) Run(operation internal.Operation, logger logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	if operation.KymaResourceNamespace == "" {
+		return step.operationManager.OperationFailed(operation, "namespace for Kyma resource not specified", nil, logger)
+	}
+
+	kymaUnstructured := &unstructured.Unstructured{}
+	kymaUnstructured.SetGroupVersionKind(steps.KymaResourceGroupVersionKind())
+	err := step.kcpClient.Get(context.Background(), client.ObjectKey{
+		Namespace: operation.KymaResourceNamespace,
+		Name:      steps.KymaName(operation),
+	}, kymaUnstructured)
+
+	if err == nil {
+		return step.operationManager.RetryOperation(operation, "Kyma resource still exists", err, backoffForK8SOperation, timeoutForK8sOperation, logger)
```

Comment:
at the moment, the workflow is to use `RetryWithoutFail` and log the error, KEB otherwise fails the operation and keeps the cluster around. Per @PK85 and @piotrmiskiewicz and it's desired to go through all steps in deprovisioning and finally cluster deprovision even if some steps in the middle fail.

https://github.com/kyma-project/control-plane/blob/45d13603ae7b07ec1394eb7b841865b045ac60f6/components/kyma-environment-broker/internal/process/deprovision_operation.go#L112

Reply:
changed